client/state/i18n/locale-suggestions/actions.js

```diff
@@ -0,0 +1,44 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { pick } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	I18N_LOCALE_SUGGESTIONS_REQUEST,
+	I18N_LOCALE_SUGGESTIONS_SUCCESS,
+	I18N_LOCALE_SUGGESTIONS_FAILURE,
+} from 'state/action-types';
+
+/**
+ * Triggers a network request to get locale suggestions
+ * @returns {Function} Action thunk
+ */
+export function requestLocaleSuggestions() {
+	return dispatch => {
+		dispatch( {
+			type: I18N_LOCALE_SUGGESTIONS_REQUEST,
+		} );
+
+		return wpcom
```

Comment:
Any reason for the thunk here instead of using the data layer? Let us know if you need help walking through that.

Reply:
See https://github.com/Automattic/wp-calypso/tree/master/client/state/data-layer as a starting point. Note that we're in the process of moving to a new interface, see: p4TIVU-8LU-p2

Other helpful folks to ping would be @dmsnell @jsnajdr and @samouri if you have any questions.

Reply:
If you haven't explored this too much before, a high level idea of what the data-layer does, is what happens if we treat API requests as side effects of a redux action?

A few cool things come out of this:
- We end up with actions that remain simple objects rather than function thunks. Very handy to test locally while using with https://github.com/Automattic/wp-calypso/tree/master/client/state/console-dispatch
- If we already have enough information in the action, the reducers can optimistically update and then correct state with another dispatched action if the server happens to disagree.
- We have a single entry point to manipulate any data that comes back from a given endpoint
- Since we're also describing http requests as actions, we can also decide whether or not to really make the requests, letting us do neat things like dedupe multiple requests, or perform automatic retrys

Reply:
Thanks for the detailed comments as always @gwwar. 

There was no reason for returning the thunk other than blindly following existing code. 
I'll migrate everything over to the data layer on this PR, and create a new one for the integration.